### PR TITLE
Burning Barrels

### PR DESCRIPTION
**Pull request by west3436**
## What this does

Allows barrels to be set on fire if they contain valid bunsen burner fuel sources. 
- Fuel is consumed at the same rate as bunsen burners.
- Wooden barrels cannot be ignited.
- You cannot drag mobs into a barrel while it is on fire.
- If you attempt to leave a barrel while it is on fire, you will take burn damage and be set on fire.
- Snuff a fire out by using your open hand on it (fucking metal innit).

image

## Why it's good

Hobo aesthetic

## Changelog

:cl:
 * rscadd: Barrels can now be set on fire.

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
Finally

**Comment by Optimism333** [comment]
Good stuff

**Comment by Eneocho** [comment]
Maintenance goblin bros... It is our time.

**Comment by PrimeDSS13** [comment]
haha YES YESSS

**Comment by Hinaichigo** [comment]
Nice. Some suggestions:

- you shouldn't be able to put it out with your hand
- you should be able to cook on it similar to a grill or a bonfire.
- you should be able to put flammable stuff into it like paper and cardboard boxes and chunks of wood to keep the fire going (this one is maybe something to consider later)

**Comment by Eneocho** [comment]
>you should be able to put flammable stuff into it like paper and cardboard boxes and chunks of wood to keep the fire going 

THIS.
Could also interact with the smoke PR to add a single tile (where the barrel is) of smoke every now and then if burning wood/paper/cardboard instead of proper fuel.

**Comment by Cartlord** [comment]
Why not let us shove people into burning barrels as an execution method?

**Comment by west3436** [comment]
> Nice. Some suggestions:
> 
> * you shouldn't be able to put it out with your hand
> * you should be able to cook on it similar to a grill or a bonfire.
> * you should be able to put flammable stuff into it like paper and cardboard boxes and chunks of wood to keep the fire going (this one is maybe something to consider later)

Will implement the first two points. As for the third, these fires are liquid-fueled rather than solid fueled. Trying to make it do both would be needlessly complex.

**Comment by Eneocho** [comment]
> Will implement the first two points. As for the third, these fires are liquid-fueled rather than solid fueled. Trying to make it do both would be needlessly complex.

You could add a timer to go on top of the liquid fuel tick which only increases if solid fuels were used.
Something like:`
var/solidfuel = 0`
then add if(/obj/item/stack/sheet/wood) to the attackby, increasing solidfuel by 50, then
```
if(solidfuel > 0)
   consumptionrate = 0
   solidfuel = (solidfuel-1)
```

File: code/modules/reagents/reagent_dispenser.dm [hunk 1]
```diff
@@ -525,6 +525,37 @@
 	flags = FPRINT | TWOHANDABLE | MUSTTWOHAND // If I end up being coherent enough to make it holdable in-hand
 	var/list/exiting = list() // Manages people leaving the barrel
 	health = 50
+	var/burning = FALSE
+	var/list/comfyfire = list('sound/misc/comfyfire1.ogg','sound/misc/comfyfire2.ogg','sound/misc/comfyfire3.ogg',)
+	var/list/possible_fuels = list(
```
Comment (Kurfursten) [inline code/modules/reagents/reagent_dispenser.dm]: instead of pasting, make this a global list and call it in both places; more maintainable

Comment (west3436) [reply code/modules/reagents/reagent_dispenser.dm]: Implemented in latest commit

Comment (Kurfursten) [reply code/modules/reagents/reagent_dispenser.dm]: you did it with the comfy fire sounds but not the much larger fuel list

Comment (west3436) [reply code/modules/reagents/reagent_dispenser.dm]: ah, didn't realize you were referring to both. will fix in next commit

**Pull request merged**